Microsoft.Toolkit.Uwp/Helpers/DispatcherHelper.cs

```diff
@@ -165,7 +165,16 @@ public static Task<T> AwaitableRunAsync<T>(this CoreDispatcher dispatcher, Func<
              {
                  try
                  {
-                     taskCompletionSource.SetResult(await function().ConfigureAwait(false));
+                     var awaitableResult = function();
+                     if (awaitableResult != null)
+                     {
+                         var result = await awaitableResult.ConfigureAwait(false);
+                         taskCompletionSource.SetResult(result);
+                     }
+                     else
+                     {
+                         taskCompletionSource.SetResult(default(T));
```

Comment:
I would propose to call taskCompletionSource.SetException(new InvalidOperationException("Task returned from async function parameter cannot be null.")); Make the dev aware that he (accidentially) misused the helper.